test/create-article.ts

```diff
@@ -1,14 +1,26 @@
+import { namedNode } from '@rdfjs/data-model';
 import deepFilter from 'deep-filter';
+import { Options } from 'jsonld';
 import { JsonLdObj } from 'jsonld/jsonld-spec';
-import { BlankNode } from 'rdf-js';
+import { BlankNode, NamedNode } from 'rdf-js';
 import { termToString } from 'rdf-string';
 
 const isNotUndefined = (arg: unknown): boolean => arg !== undefined;
 
-export default (id?: BlankNode, name = id ? `Article ${id}` : 'Article'): JsonLdObj => (
+type Options = {
+  id?: BlankNode;
+  name?: string;
+  types?: Array<NamedNode>;
+}
+
+export default ({
+  id = undefined,
+  name = 'Article',
+  types = [namedNode('http://schema.org/Article')],
+}: Options = {}): JsonLdObj => (
   deepFilter({
     '@id': id ? termToString(id) : undefined,
-    '@type': 'http://schema.org/Article',
+    '@type': types.map(termToString),
```

Comment:
why passing in `NamedNode` instances if they have to be cast down to string anyway? Any validation happening in `namedNode()`?

Reply:
Yep, it needs to be an IRI.